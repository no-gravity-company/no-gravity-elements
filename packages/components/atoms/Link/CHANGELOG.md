# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.1.0](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/link@1.0.13...@no-gravity-elements/link@1.1.0) (2024-11-12)

### Features

- new button styles ([c850785](https://github.com/no-gravity-company/no-gravity-elements/commit/c850785418de3a4f9ca393c75260ed407061314a))

## [1.0.13](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/link@1.0.12...@no-gravity-elements/link@1.0.13) (2024-03-30)

**Note:** Version bump only for package @no-gravity-elements/link

## [1.0.12](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/link@1.0.8...@no-gravity-elements/link@1.0.12) (2024-03-29)

### Bug Fixes

- Ignore changelogs in lerna config ([9f9c7ee](https://github.com/no-gravity-company/no-gravity-elements/commit/9f9c7ee07e4e05f8dfe9c934bf884515ee8d0732))
