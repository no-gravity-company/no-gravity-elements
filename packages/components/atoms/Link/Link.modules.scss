@import '../../../stylesheets/base/variables';
@import '../../../stylesheets/base/colors';

a {
  text-decoration: none;
  position: relative;
  &:hover::after {
    content: '';
    display: block;
    width: 100%;
    height: baseUnit(0.125);
    background-color: var(--color, #{$primary-color});
    position: absolute;
    bottom: baseUnit(-0.25);
    left: 0;
    border-radius: baseUnit(0.75);
  }
}
