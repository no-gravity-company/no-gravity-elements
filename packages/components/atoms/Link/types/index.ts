import { TypographySizes } from '@no-gravity-elements/types';

export interface LinkProps {
  href?: string;
  size?: TypographySizes;
}
