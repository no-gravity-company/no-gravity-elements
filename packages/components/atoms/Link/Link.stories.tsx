import { Story } from '@storybook/preact';
import { html } from 'lit-html';
import { TypographySizes } from 'packages/types';

import { LinkProps } from '@atoms/Link/types';

import '@no-gravity-elements/link';

export default {
  title: 'Link',
  component: 'nge-link',
  parameters: {
    webComponents: {
      tagName: 'nge-link',
    },
  },
  argTypes: {
    text: {
      description: 'Link text',
      control: { type: 'text' },
      table: {
        defaultValue: { summary: 'string' },
      },
    },
    size: {
      description: 'Text size',
      control: 'select',
      options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'h7', 'sp', 'p', 'lp', 'legal', 'button'],
    },
    href: {
      description: 'Link url',
      control: { type: 'text' },
      table: {
        defaultValue: { summary: '/' },
      },
    },
  },
};
type StoryLinkProps = LinkProps & { text: string };

export const Default: Story<StoryLinkProps> = (args): any => {
  const { text, href, size } = args;

  return html` <nge-link size="${size}" href="${href}">${text}</nge-link> `;
};

Default.args = {
  href: '/',
  text: 'Link',
  size: TypographySizes.h6,
};
