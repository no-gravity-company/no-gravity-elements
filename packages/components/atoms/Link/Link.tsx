import { Fragment, h } from 'preact';

import { LinkProps } from '@atoms/Link/types';

import { CustomFunctionComponent } from '@types';

import '@no-gravity-elements/typography';
/**
 * <nge-link>
 *
 *  Link component
 *
 * @element nge-link
 *
 * @prop {'string} [href] - link url
 * @prop {h1|h2|h3|h4|h5|h6|h7|sp|p|lp|legal|button} [size] - Size of the text
 *
 * @example
 * <nge-link href="/">Link</nge-link>
 */

const Link: CustomFunctionComponent<LinkProps> = ({ href, size }: LinkProps) => {
  return (
    <Fragment>
      <a href={href}>
        <nge-typography size={size}>
          <slot />
        </nge-typography>
      </a>
    </Fragment>
  );
};

Link.observedAttributes = ['size', 'href'];

export default Link;
