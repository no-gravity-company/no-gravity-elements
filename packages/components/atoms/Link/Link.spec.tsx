import { h } from 'preact';
import { shallow } from 'enzyme';

import Link from '@atoms/Link/Link';

import { TypographySizes } from '@types';

describe('Link', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Link size={TypographySizes.h1} href='/'>
        Link
      </Link>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it.each`
    size
    ${TypographySizes.h1}
    ${TypographySizes.h4}
    ${undefined}
  `('should render correctly with the provided size', ({ size }) => {
    const wrapper = shallow(
      <Link size={size} href='/'>
        Link
      </Link>,
    );
    const element = wrapper.find(size || TypographySizes.p);
    expect(element).not.toBe(null);
  });
});
