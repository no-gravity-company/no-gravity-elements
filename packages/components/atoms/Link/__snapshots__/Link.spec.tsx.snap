// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match the snapshot 1`] = `
<a
  href="/"
>
  <nge-typography
    size="h1"
  >
    <slot />
  </nge-typography>
</a>
`;
